### PR TITLE
interp: Fix panic when trying to `read` and stdin is `nil`

### PR DESCRIPTION
**Pull request by andreynering**

### CODE REVIEW

File: interp/builtin.go [hunk 1]
```diff
@@ -719,6 +720,11 @@ func (r *Runner) printOptLine(name string, enabled bool) {
 }
 
 func (r *Runner) readLine(raw bool) ([]byte, error) {
+	if r.stdin == nil {
+		r.errf("can't read, there's no stdin\n")
```
Comment (mvdan) [inline interp/builtin.go:724]: if we're returning an error, I don't think we want to print one as well

File: interp/interp_test.go [hunk 2]
```diff
@@ -3548,3 +3548,21 @@ func TestMalformedPathOnWindows(t *testing.T) {
 		t.Fatalf("wrong output:\nwant: %q\ngot:  %q", want, got)
 	}
 }
+
+func TestReadShouldNotPanicForEmptyStdin(t *testing.T) {
```
Comment (mvdan) [inline interp/interp_test.go:3552]: s/ForEmptyStdin/WithNilStdin/ maybe?

File: interp/interp_test.go [hunk 3]
```diff
@@ -3548,3 +3548,21 @@ func TestMalformedPathOnWindows(t *testing.T) {
...
+
+	f := parse(t, nil, "read foobar")
+	if err := r.Run(context.Background(), f); err == nil {
+		t.Fatal("it should have retuned an error")
+	}
+
+	want := "can't read, there's no stdin\n"
+	if got := b.String(); got != want {
+		t.Fatalf("wrong output:\nwant: %q\ngot:  %q", want, got)
+	}
```
Comment (mvdan) [inline interp/interp_test.go:3567]: if you remove the error printing, then this part of the test can go as well

File: interp/builtin.go (hunk 1 above)
Comment (andreynering) [reply interp/builtin.go]: Thanks for the blazing fast review!

**Review by mvdan — APPROVED** [review]
Thanks for the fix!

**Comment by mvdan** [comment]
Sigh, it appears one of the mac machines got stuck.

**Pull request merged**